### DISCUSSION TITLE
How can the data in the VTK pipeline be dynamically obtained when the train is started? The files read by the current demo are all dead. How can the files be dynamically obtained

### DISCUSSION THREAD

**Discussion by eagle201306** [Kitware/trame#104] [Trame/VTK]
How can the data in the VTK pipeline be dynamically obtained when the train is started? The files read by the current demo are all dead. How can the files be dynamically obtained

**Comment by jourdain** [comment]
Sorry I don't understand the question.

**Reply by eagle201306** [reply]
My requirement is the server When the start method starts the server, pass in the parameters. During VTK rendering, read the file path I passed in and render. How do I need to implement it

**Reply by eagle201306** [reply]
from trame.app import get_server
from trame.ui.vuetify import SinglePageLayout
from trame.widgets import vtk, vuetify

from vtkmodules.vtkFiltersSources import vtkConeSource
from vtkmodules.vtkRenderingCore import (
    vtkActor,
    vtkPolyDataMapper,
    vtkRenderer,
    vtkRenderWindow,
    vtkRenderWindowInteractor,
)

from vtkmodules.vtkInteractionStyle import vtkInteractorStyleSwitch  # noqa


import vtkmodules.vtkRenderingOpenGL2  # noqa



# VTK pipeline


renderer = vtkRenderer()
renderWindow = vtkRenderWindow()
renderWindow.AddRenderer(renderer)

renderWindowInteractor = vtkRenderWindowInteractor()
renderWindowInteractor.SetRenderWindow(renderWindow)
renderWindowInteractor.GetInteractorStyle().SetCurrentStyleToTrackballCamera()

cone_source = vtkConeSource()
mapper = vtkPolyDataMapper()
mapper.SetInputConnection(cone_source.GetOutputPort())
actor = vtkActor()
actor.SetMapper(mapper)

renderer.AddActor(actor)
renderer.ResetCamera()


server = get_server()
ctrl = server.controller

with SinglePageLayout(server) as layout:
    layout.title.set_text("Hello trame")

    with layout.content:
        with vuetify.VContainer(
            fluid=True,
            classes="pa-0 fill-height",
        ):
            view = vtk.VtkLocalView(renderWindow)
            ctrl.on_server_ready.add(view.update)



if __name__ == "__main__":
    server.start()


**example In Code：cone_source    How to read files dynamically instead**

**Reply by jourdain** [reply]
How do you want to pass that file path?

**Reply by jourdain** [reply]
from URL at startup in multi-user or dynamically while the application is running?

**Reply by eagle201306** [reply]
How to use multi-user startup, is there an example

**Reply by eagle201306** [reply]
> How do you want to pass that file path?

Yes, we have found a solution. Thank you

**Reply by jourdain** [reply]
Docker make the multi-user setup straight forward. The [cookiecutter](https://github.com/Kitware/trame-cookiecutter) provide the base for that setup. 
Documentation explaining the principle is available [here](http://kitware.github.io/paraviewweb/docs/multi_user_setup.html)

**Reply by eagle201306** [reply]
OK, let me take a look at the relevant introduction first. Thank you
